### DISCUSSION TITLE
WXT

### DISCUSSION THREAD

**Discussion by Malix-Labs** [uBlockOrigin/uBOL-home#237] [Q&A] [answered]
Hey team,

Have you thought about using [WXT](https://wxt.dev/) instead of having to duplicate the code between chromium and quantum ?

I mention it because it helped me to do just that while keeping a minimalist approach to code

It's a vite plugin that does the work you would have to do yourself anyway, basically

I helped to build [its comparison page](https://wxt.dev/guide/resources/compare.html) btw, and can attest it is accurate

**Comment by gorhill** [comment, marked as answer]
There is no code duplicated, Edge's version is essentially Chromium's version with a few changes required by the Edge store, performed by [a shell script](https://github.com/uBlockOrigin/uBOL-home/blob/main/tools/make-edge.sh).

**Reply by Malix-Labs** [reply]
Is there not duplication between Chromium and Quantum (Firefox) ?

**Reply by gorhill** [reply]
Same [extension code](https://github.com/gorhill/uBlock/tree/master/platform/mv3/extension), only difference is the [manifest.json file](https://github.com/gorhill/uBlock/tree/master/platform/mv3/firefox) and [a background page](https://github.com/gorhill/uBlock/blob/master/platform/mv3/firefox/background.html) to load the [JS file](https://github.com/gorhill/uBlock/blob/master/platform/mv3/extension/js/background.js) used as worker in Chromium version.
